istio-proxy/common.sh

```diff
@@ -1,8 +1,11 @@
-set -x 
+set -x
 set -e
 
+#if the user has Bazelisk installed, then USE_BAZEL_VERSION will fix the bazel
+#version. If not, error out.
+export USE_BAZEL_VERSION=1.1.0
```

Comment:
this will override a `.bazelversion` file in the repo. do we really want that? we'll have to keep updating this version

Reply:
I'm not convinced we should use bazelisk in a build environment, tbh (won't work at all in brew).

I'll need to check, but we used to have bazel as a build dependency in copr, pinned to a specific version. This should enable to use w/e is in copr.

Reply:
You're right, bazelisk pulls down things on the fly, that won't work in brew. Let's remove this part for now

Reply:
I can remove this, but: 

* in regards to COPR and Brew, these are using Bazel, not Bazelisk, so this environment variable will be ignored on them and will have *no* effect. 
* running locally, the Bazel version changes very often meaning that different builds on the same system will likely require different Bazel versions. 
* the errors for different Bazel versions are often very esoteric and not indicative of running the wrong Bazel version
* There are two stages to the build, one of which runs locally on whatever version of Bazel that you have to generate the source rpm, the other of which runs on COPR/Brew to compile Envoy.
*  `.bazelversion` only exists in the compile stage. We could copy this over to the rpms repo and use that instead of this environment variable for the RPM builds. I could also move the environment variable over to fetch.sh and then it won't even be present in Brew (although again, Bazel doesn't care about the variable/file, only Bazelisk. Bazel will just either succeed with the wrong version of Bazel or fail with esoteric errors)